### PR TITLE
bluetooth: controller: kconfig: Add periodic adv event length kconfig

### PR DESCRIPTION
**Pull request by sean-madigan**
This can be used to set the periodic advertising event length via sdc_hci_cmd_vs_periodic_adv_event_length_set

### CODE REVIEW

File: subsys/bluetooth/controller/hci_driver.c [hunk 1]
```diff
@@ -490,6 +490,14 @@ static int configure_supported_features(void)
 		if (err) {
 			return -ENOTSUP;
 		}
+
+		sdc_hci_cmd_vs_periodic_adv_event_length_set_t params = {
+			.event_length_us = CONFIG_BT_CTLR_SDC_PERIODIC_ADV_EVENT_LEN_DEFAULT
```
Comment (rugeGerritsen) [inline subsys/bluetooth/controller/hci_driver.c:495]: Do we have to change something inside the lib to ensure this value is used after HCI reset?

Comment (sean-madigan) [reply subsys/bluetooth/controller/hci_driver.c:495]: That's a good point

File: subsys/bluetooth/controller/Kconfig [hunk 2]
```diff
@@ -138,6 +138,19 @@ config BT_CTLR_SDC_MAX_CONN_EVENT_LEN_DEFAULT
 	  microseconds. The event length and the connection interval are the
 	  primary parameters for setting the throughput of a connection.
 
+config BT_CTLR_SDC_PERIODIC_ADV_EVENT_LEN_DEFAULT_OVERRIDE
```
Comment (hermabe) [inline subsys/bluetooth/controller/Kconfig:141]: What is the reason for having the `_OVERRIDE` config? Can we not just set the default for `BT_CTLR_SDC_PERIODIC_ADV_EVENT_LEN_DEFAULT` unconditionally?

Comment (rugeGerritsen) [reply subsys/bluetooth/controller/Kconfig:141]: We want to be able override the default out of tree for certain configurations

Comment (hermabe) [reply subsys/bluetooth/controller/Kconfig:141]: That should be possible anyway as long as the out-of-tree config definition is found before this one. Though maybe simpler to keep it for now.

Comment (rugeGerritsen) [reply subsys/bluetooth/controller/Kconfig:141]: > That should be possible anyway as long as the out-of-tree config definition is found before this one.

Unfortunately this is not always the case

File: subsys/bluetooth/controller/hci_driver.c [hunk 3]
```diff
@@ -490,6 +490,14 @@ static int configure_supported_features(void)
...
+		};
+		err = sdc_hci_cmd_vs_periodic_adv_event_length_set(&params);
```
Comment (seko-nordic) [inline subsys/bluetooth/controller/hci_driver.c:497]: I'm not sure it's good to call HCI commands before sdc_enable. Can we move this call to hci_driver_open instead?

File: subsys/bluetooth/controller/hci_driver.c [hunk 4]
```diff
@@ -965,6 +965,17 @@ static int hci_driver_open(void)
 		return err;
 	}
```
Comment (seko-nordic) [inline subsys/bluetooth/controller/hci_driver.c:967]: can you also set normal event length here using VS cmd and remove corresponding call to sdc_cfg_set(SDC_CFG_TYPE_EVENT_LENGTH)
That would be a nice unification / cleanup

**Review by seko-nordic — APPROVED** [review]

**Pull request merged**